---
title: "Molecular & Genomic Epidemiology of Infectious Diseases: Lessons & Directions After COVID-19"
subtitle: "<hr>"
author: "Brendan J. Kelly, MD, MS"
institute: "Infectious Diseases & Epidemiology<br><br>![](penn_logos/Penn-Shield-Logotype-White-over-Blue-RGB-Expanded.svg){width=160px}" 
date: "`r format(Sys.Date(), '%d %B %Y')`"
output:
  xaringan::moon_reader:
    css: xaringan-themer-inferno-grids-fade-after.css
    lib_dir: libs
    chakra: libs/remark-latest.min.js
    seal: false
    nature:
      ratio: 16:9
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
      slideNumberFormat: "%current%"

---
exclude: true

```{r setup, include=FALSE, echo=FALSE, message=FALSE, error=FALSE, include=FALSE, comment=FALSE}
options(htmltools.dir.version = FALSE)
xaringan::summon_remark()
library(tidyverse)
library(gt)

```

```{r xaringan-tile-view, echo=FALSE, message=FALSE, error=FALSE, include=FALSE, comment=FALSE}
xaringanExtra::use_tile_view()
xaringanExtra::use_webcam()
xaringanExtra::use_tachyons()
xaringanExtra::use_animate_css()
xaringanExtra::use_share_again()
#xaringanExtra::use_scribble(pen_color = "#56106E")
```

---
class: top left hide-count
background-image: url(img/sarscov2_cdc_blur_crop.png)
background-size: cover

.move-top10[
.title-text[<span style='font-size:0.6em'>Molecular & Genomic<br> Epidemiology<br> of Infectious Diseases</span>]  
<hr>
.title-subtext[<span style='font-size:1.0em'>Lessons and Directions<br>After COVID-19</span>]  
]  
  
.callout-url-leftcorner[
.title-nametext[
Brendan J. Kelly, MD, MS  
Infectious Diseases, Epidemiology & Microbiology  
University of Pennsylvania  
14 November 2023  
<!-- `r format(Sys.Date(), '%d %B %Y')`    -->
]
]

.footnote-right[<span style='color: white'>photo: CDC @unsplash</span>]  



---

## Disclosures  

.pad-left[  

- No conflicts of interest.  

- Research supported by:

    - NIAID K23 AI121485  
    
    - CDC BAA 200-2021-10986   

]



---
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-a animated flipInY"><div>COVID-19<br>Genomic<br>Epidemiology</div></div>
</div>


---
layout: false
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-a"><div>COVID-19<br>Genomic<br>Epidemiology</div></div>
<div class="extension-tile gridset-b animated flipInX"><div>Biases<br>Limitations<br>& Unknowns</div></div>
</div>


---
layout: false
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-a"><div>COVID-19<br>Genomic<br>Epidemiology</div></div>
<div class="extension-tile gridset-b"><div>Biases<br>Limitations<br>& Unknowns</div></div>
<div class="extension-tile gridset-c animated bounceInDown"><div>COVID-19<br>Aftermath</div></div>
</div>



---
layout: false
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-a"><div>COVID-19<br>Genomic<br>Epidemiology</div></div>
<div class="extension-tile gridset-grey"><div>Biases<br>Limitations<br>& Unknowns</div></div>
<div class="extension-tile gridset-grey"><div>COVID-19<br>Aftermath</div></div>
</div>




---

## Genomic Epidemiology        
  
.pad-left[  
  
- __Epidemiology__:    
  
    - study of the distribution and determinants of health-related outcomes in a specified population  

- __Molecular Epidemiology__:     
  
    -  joins understanding of disease at the molecular level with population-based study designs and approaches; links epidemiology with laboratory sciences  
    
- __Genomic Epidemiology__:     
  
    -  use of pathogen genomic data to determine the distribution and spread of an infectious disease in a specified population  
      
]  

.footnote-left[[CDC COVID-19 Genomic Epidemiology Toolkit 1.1](https://www.cdc.gov/amd/pdf/slidesets/toolkitmodule_1.1-508c.pdf); Lash TL et al _Modern Epidemiology (4th ed)_]  
  
  
  
---

## SARS-CoV-2 Genomic Epidemiology        
  
.pad-left[  
  
- __SARS-CoV-2 Genomes__:    
  
    - SARS-CoV-2 genome: ~ 30,000 nucleotides (~ 10,000 amino acids)  
    
    - infer relatedness from phylogenetic distance    
    
- __RNA Genome Sequencing__:     
  
    - extract RNA, reverse transcription, multiplex PCR, end repair & ligation indexing  
    
    - short-read or long-read sequencing  
  
]  

.footnote-left[[CDC COVID-19 Genomic Epidemiology Toolkit 1.1](https://www.cdc.gov/amd/pdf/slidesets/toolkitmodule_1.1-508c.pdf)]  
  
  
  
---
  
## National & Local Applications    
  
.pad-left[  
  
- **National**:  

    - monitor emergence and movement of new strains   
  
    - monitor trends after intervention (e.g., vaccination)   
  
- **Local**:   
    
    - investigate clusters for transmission (workplace, healthcare, etc)    
    
    - reveal unexpected clusters    
    
]  


.footnote-left[[CDC COVID-19 Genomic Epidemiology Toolkit 1.1](https://www.cdc.gov/amd/pdf/slidesets/toolkitmodule_1.1-508c.pdf); [Washington Post](https://www.washingtonpost.com/graphics/2020/health/coronavirus-genetic-code/)]  

  
  
---
exclude: false

## International Collaboration      
  
.center[
```{r gisaid_website, echo=FALSE, message=FALSE, error=FALSE, comment=FALSE, eval=TRUE, autodep=TRUE, fig.align='center', out.width='95%'}
knitr::include_url("https://gisaid.org/")
```
]

  
.footnote-left[https://gisaid.org/]  

  
---
exclude: false

## International Collaboration      
  
.center[
```{r nextrain_website, echo=FALSE, message=FALSE, error=FALSE, comment=FALSE, eval=TRUE, autodep=TRUE, fig.align='center', out.width='60%'}
knitr::include_graphics("figs/nextstrain_ncov_screenshot.png")
```
]

  
.footnote-left[https://nextstrain.org/ncov/open/global/all-time]  

  

---

## Variants of Concern  

.center[
```{r cdc_website, echo=FALSE, message=FALSE, error=FALSE, comment=FALSE, eval=TRUE, autodep=TRUE, fig.align='center', out.width='95%'}
knitr::include_url("https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-classifications.html")
```
]

  
.footnote-left[https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-classifications.html]  

  

---
exclude: false

## Local Tracking        
  
.center[
```{r penn_website, echo=FALSE, message=FALSE, error=FALSE, comment=FALSE, eval=TRUE, autodep=TRUE, fig.align='center', out.width='95%'}
knitr::include_url("https://microb120.med.upenn.edu/data/SARS-CoV-2")
```
]

  
.footnote-left[https://microb120.med.upenn.edu/data/SARS-CoV-2]  
  


---
layout: false
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-grey"><div>COVID-19<br>Genomic<br>Epidemiology</div></div>
<div class="extension-tile gridset-b"><div>Biases<br>Limitations<br>& Unknowns</div></div>
<div class="extension-tile gridset-grey"><div>COVID-19<br>Aftermath</div></div>
</div>


---

## Biases: SARS-CoV-2 "Viral Load"    
  
.pad-left[  
  
- The number of virions detectable in respiratory secretions varies between individuals and over the course of infection:    
  
    - high viral loads are required for most sequencing-preparation pipelines    
    
    - coverage/quality may be poor if viral loads are low    
  
- What are the consequences of excluding low-viral-load samples from genomic epidemiology surveillance?  
    
]    


---

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs//p_ct_distribution_ridgeplot_flip.png")

```

]



---

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs/p_ct_distribution_ridgeplot_with_seqbar_flip.png")

```

]


---

## Limitations: SARS-CoV-2 vs Population Immunity    
  
.pad-left[  
  
- Molecular evolution occurs in the context of increasingly complex and heterogeneous population immunity:    
  
    - waves of SARS-CoV-2 variants with different immune epitopes    
    
    - the introduction of vaccines    
  
- How best to measure SARS-CoV-2 molecular evolution in this context? (e.g., how best to perform "antigenic cartography"?)  
    
]    

.footnote-left[Mykytyn AZ et al _Science Immunology_ 2022]  


---

## SARS-CoV-2 Molecular Evolution    
  
.pad-left[  
  
- How do we measure SARS-CoV-2 genome change?  
  
    - daily, weekly, monthly?    
    
    - nucleotide, dN/dS, AA, gene?
    
    - how to pool within codons, genes?    

    - covariance across genome?

- How does genomic change relate to changes in incidence?  


    
]  


---

## Genomic Positional Diversity

.pad-left[

- Shannon diversity: 
    
  $$H' = - \sum{ p_{i} * \log_{b}{(p_{i})} }$$
    
<center>(note: typically natural log or base 2 are used)</center>

- richness: how many nucleotide variants are there?
    
- evenness: how are variants distributed?


]


---

## Aside on Information Theory

.pad-left[

- Shannon diversity:

    $$H' = - \sum{ p_{i} * \log_{b}{(p_{i})} }$$
    
- Claude Shannon & information entropy:

    $$H(p) = - \sum{ p_{i} * \log_{b}{(p_{i})} }$$
    
- **"The uncertainty contained in a probability distribution is the average log-probability of an event."**

]

.footnote[McElreath _Statistical Rethinking, 2nd_ 2020]


---
exclude: true
background-image: url(figs/scv2_snv_diversity_per_position_and_time_clear.gif)
background-size: contain


---
exclude: true
background-image: url(figs/scv2_snv_diversity_per_position_and_time_shadow.gif)
background-size: contain


---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/scv2_snv_diversity_per_position_and_time_clear_color.gif")

```


---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/scv2_snv_diversity_per_position_and_time_shadow_color.gif")

```


---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/scv2_peptide_diversity_per_position_and_time_clear.gif")

```


---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/scv2_peptide_diversity_per_position_and_time_shadow.gif")

```

---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/p_phl_cases.png")

```



---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/p_lm_logdiff_peptide_color.png")

```


---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/p_lm_logdiff_peptide_color_tag.png")

```



---
exclude: false

## The Canyon Hypothesis    
  
.pad-left[

- SARS-CoV-2 molecular evolution & the "Canyon Hypothesis":

    - highly conserved cell entry mechanism mediated by the spike protein (S gene)

    - spike engages angiotensin converting enzyme 2 (ACE2) & host proteases enable efficient membrane fusion between virions and target cells

    - a model by which sarbecoviruses are activated for fusion competency and interplay between humoral immunity and the molecular evolution

]


.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]




---
exclude: false

## The Canyon Hypothesis    
  
.pad-left[

- during circulation in populations with high rates of humoral immunity, viral entry proteins favor predominantly closed RBD configurations

- immediately after spillover into a population that lacks immunity, the newly emergent virus remains closely related to its ancestor

- during sustained transmission between seronegative individuals, wide transmission bottlenecks facilitate rapid emergence of variants that favor open RBD configurations to spread rapidly

- as humoral immunity expands, it gradually leads to a return to closed RBDs as repeat exposures facilitate the affinity maturation of expansive antibody repertoires that are disproportionately costly to open RBD configurations

]


.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]




---

## The Canyon Hypothesis    

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs/canyon_hypothesis_rbd_open_closed.png")

```

]

.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]


---
exclude: false

## The Canyon Hypothesis    
  
.pad-left[

- most changes in S prior to the emergence of the Omicron variant appear to have been driven by selection for improved transmission between immunologically naive hosts

- S:D614G spike mutation improved ACE2 affinity but also made RBDs were more likely to hold the open conformation

- cryo-EM studies of the Omicron S protein suggest that while the Delta Spike predominantly occupies conformations with 1 or more RBDs open simultaneously, the Omicron Spike appears to prefer conformations with 0 or 1 open RBD

- antigenicity of stabilizing elements & tendency of primary immune responses to generate a limited repertoire of antibodies may explain the selection for open RBDs early in the pandemic & shift in the selective landscape that led to the Omicron variant’s emergence and rapid sweep

]


.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]



---
exclude: true

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs//p_ct_distribution_ridgeplot_flip.png")

```

]



---
exclude: true

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs/p_ct_distribution_ridgeplot_with_seqbar_flip.png")

```

]



---

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs/alpha_delta_omicron_ct_comparison.png")

```

]



---

## Unknowns: Where did 'weird' Omicron come from?

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs/new_long_branch_screenshot.png")

```

]

.footnote-left[[Kupferschmidt K *Science* 1 Dec 2021](https://www.science.org/content/article/where-did-weird-omicron-come)]  

---

## Unknowns: Where did 'weird' Omicron come from?

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs/three_explanations_screenshot.png")

```

]

.footnote-left[[Kupferschmidt K *Science* 1 Dec 2021](https://www.science.org/content/article/where-did-weird-omicron-come)]  




---

## Pennsylvania Deer

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs/mbio.02101-22-f005.gif")

```

]

.footnote-left[[Marques AD et al *mBio* 2022](http://dx.doi.org/10.1128/mbio.02101-22)]  




---
layout: false
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-grey"><div>COVID-19<br>Genomic<br>Epidemiology</div></div>
<div class="extension-tile gridset-grey"><div>Biases<br>Limitations<br>& Unknowns</div></div>
<div class="extension-tile gridset-c"><div>COVID-19<br>Aftermath</div></div>
</div>




---

## Lessons & Directions After COVID-19      
  
.pad-left[  
  
- What has come of this unprecedented investment in molecular/genomic epidemiology?  
  
    - unprecedented pathogen sequencing capacity    
    
    - novel methods of population surveillance: wastewater surveillance    

    - political animus against CDC & resulting budget cuts  
    
]  

---

## Ongoing SARS-CoV-2 Variant Tracking   

.center[
```{r ongoing_variants_website, echo=FALSE, message=FALSE, error=FALSE, comment=FALSE, eval=TRUE, autodep=TRUE, fig.align='center', out.width='95%'}
knitr::include_url("https://covid.cdc.gov/covid-data-tracker/#variant-proportions")
```
]

.footnote[[https://covid.cdc.gov/covid-data-tracker/#variant-proportions](https://covid.cdc.gov/covid-data-tracker/#variant-proportions)]


---

## Wastewater SARS-CoV-2 Surveillance     

.center[
```{r ongoing_wastewater_website, echo=FALSE, message=FALSE, error=FALSE, comment=FALSE, eval=TRUE, autodep=TRUE, fig.align='center', out.width='95%'}
knitr::include_url("https://covid.cdc.gov/covid-data-tracker/#wastewater-surveillance")
```
]

.footnote[[https://covid.cdc.gov/covid-data-tracker/#wastewater-surveillance](https://covid.cdc.gov/covid-data-tracker/#wastewater-surveillance)]



---

## Centers for Pathogen Genomics       

.center[
```{r pathogen_genomics_website, echo=FALSE, message=FALSE, error=FALSE, comment=FALSE, eval=TRUE, autodep=TRUE, fig.align='center', out.width='95%'}
knitr::include_url("https://www.cdc.gov/media/releases/2022/p0920-PGCoE-network.html")
```
]

.footnote[[https://www.cdc.gov/media/releases/2022/p0920-PGCoE-network.html](https://www.cdc.gov/media/releases/2022/p0920-PGCoE-network.html)]



---

## NWSS: National Wastewater Surveillance System         

.center[
```{r nwss_website, echo=FALSE, message=FALSE, error=FALSE, comment=FALSE, eval=TRUE, autodep=TRUE, fig.align='center', out.width='95%'}
knitr::include_url("https://www.cdc.gov/nwss/index.html")
```
]

.footnote[[https://www.cdc.gov/nwss/index.html](https://www.cdc.gov/nwss/index.html)]



---

## CDC Budget Cuts        

.pad-left[  
  
- CDC and NIH have become political targets:   
  
    - $1.3 billion (~18%) cut from CDC as part of debt ceiling agreement      
    
    - some budget proposals for FY2024 include a total $11.5 billion ($2.8 billion increase from FY2023)     

    - other budget proposals for FY2024 include additional 20-30% cuts  
    
]  



---

## Lessons & Directions After COVID-19      
  
.pad-left[  
  
- What has come of this unprecedented investment in molecular/genomic epidemiology?    
  
    - unprecedented pathogen sequencing capacity    
    
    - novel methods of population surveillance: wastewater surveillance    

    - political animus against CDC & resulting budget cuts  

- **Will re-investment come before or after the next pandemic?**  

    
]  


---
exclude: true

## Acknowledgements    

.pull-left[
- __ARES Group @ Penn__
Laurel Glaser, Kyle Rodino, Sean Loughrey, Laura Cowden,   
Magda Wernovsky, Jessie Gunoskey, Jennifer Han, Erik Clarke, Michael David,
Matt Ziegler, Lauren Dutcher, Ebbing Lautenbach, Jim Harrigan  

- __Bushman Laboratory @ Penn__
Arwa Abbas, Aoife Roche, Andrew Marques,
Aubrey Bailey, John Everett,
Rick Bushman

- __PennCHOP Microbiome Program__
Lisa Mattei, Casey Hofstaedter, Huanjia Zhang, Kyle Bittinger

]


.pull-right[
- __Collman Laboratory @ Penn__
Ize Imai, Aurea Simon Soro, John McGinniss, Ron Collman

- __Division of ID @ Penn__
Ian Frank, Pablo Tebas, Robert Gross, Emily Blumberg

- __Rutgers University & Penn DBEI__
Jason Roy, Arman Oganisian
  
- __CDC Prevention Epicenters__
Clifford McDonald, Alison Laufer Halpin
  
- __Funding__
<u>CDC</u>: BAAs 200-2016-91964, 200-2018-02919, 200-2021-10986,
200-2021-10986 & <u>NIAID</u>: K23 AI121485

]

.center[
### brendank@pennmedicine.upenn.edu
]


---
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-a animated flipInY"><div>COVID-19<br>Genomic<br>Epidemiology</div></div>
<div class="extension-tile gridset-b animated flipInX"><div>Biases<br>Limitations<br>& Unknowns</div></div>
<div class="extension-tile gridset-c animated bounceInDown"><div>COVID-19<br>Aftermath</div></div>
</div>




---
class: middle center hide-count
background-image: url(img/sarscov2_cdc_blur_crop.png)
background-size: cover

.title-subtext[Questions?]



---
exclude: true
class: middle center hide-count
background-image: url(img/sarscov2_cdc_blur_crop.png)
background-size: cover

.title-subtext[Questions?]

.callout-url-bottom[  
.fade-in[<span style="font-size:0.5em; color:white">slides ↓ </span>]  
<span style="link-color:white">[bjklab.org](http://www.bjklab.org)</span>  
]  


