<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Molecular &amp; Genomic Epidemiology of Infectious Diseases: Lessons &amp; Directions After COVID-19</title>
    <meta charset="utf-8" />
    <meta name="author" content="Brendan J. Kelly, MD, MS" />
    <meta name="date" content="2023-11-14" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/tile-view/tile-view.css" rel="stylesheet" />
    <script src="libs/tile-view/tile-view.js"></script>
    <script src="libs/xaringanExtra-webcam/webcam.js"></script>
    <script id="xaringanExtra-webcam-options" type="application/json">{"width":"200","height":"200","margin":"1em"}</script>
    <link href="libs/tachyons/tachyons.min.css" rel="stylesheet" />
    <link href="libs/animate.css/animate.xaringan.css" rel="stylesheet" />
    <script src="libs/clipboard/clipboard.min.js"></script>
    <link href="libs/shareon/shareon.min.css" rel="stylesheet" />
    <script src="libs/shareon/shareon.min.js"></script>
    <link href="libs/xaringanExtra-shareagain/shareagain.css" rel="stylesheet" />
    <script src="libs/xaringanExtra-shareagain/shareagain.js"></script>
    <link rel="stylesheet" href="xaringan-themer-inferno-grids-fade-after.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">

exclude: true





---
class: top left hide-count
background-image: url(img/sarscov2_cdc_blur_crop.png)
background-size: cover

.move-top10[
.title-text[&lt;span style='font-size:0.6em'&gt;Molecular &amp; Genomic&lt;br&gt; Epidemiology&lt;br&gt; of Infectious Diseases&lt;/span&gt;]  
&lt;hr&gt;
.title-subtext[&lt;span style='font-size:1.0em'&gt;Lessons and Directions&lt;br&gt;After COVID-19&lt;/span&gt;]  
]  
  
.callout-url-leftcorner[
.title-nametext[
Brendan J. Kelly, MD, MS  
Infectious Diseases, Epidemiology &amp; Microbiology  
University of Pennsylvania  
14 November 2023  
&lt;!-- 14 November 2023    --&gt;
]
]

.footnote-right[&lt;span style='color: white'&gt;photo: CDC @unsplash&lt;/span&gt;]  



---

## Disclosures  

.pad-left[  

- No conflicts of interest.  

- Research supported by:

    - NIAID K23 AI121485  
    
    - CDC BAA 200-2021-10986   

]



---
class: full-screen hide-count

&lt;div class="grid-3-1"&gt;
&lt;div class="extension-tile gridset-a animated flipInY"&gt;&lt;div&gt;COVID-19&lt;br&gt;Genomic&lt;br&gt;Epidemiology&lt;/div&gt;&lt;/div&gt;
&lt;/div&gt;


---
layout: false
class: full-screen hide-count

&lt;div class="grid-3-1"&gt;
&lt;div class="extension-tile gridset-a"&gt;&lt;div&gt;COVID-19&lt;br&gt;Genomic&lt;br&gt;Epidemiology&lt;/div&gt;&lt;/div&gt;
&lt;div class="extension-tile gridset-b animated flipInX"&gt;&lt;div&gt;Biases&lt;br&gt;Limitations&lt;br&gt;&amp; Unknowns&lt;/div&gt;&lt;/div&gt;
&lt;/div&gt;


---
layout: false
class: full-screen hide-count

&lt;div class="grid-3-1"&gt;
&lt;div class="extension-tile gridset-a"&gt;&lt;div&gt;COVID-19&lt;br&gt;Genomic&lt;br&gt;Epidemiology&lt;/div&gt;&lt;/div&gt;
&lt;div class="extension-tile gridset-b"&gt;&lt;div&gt;Biases&lt;br&gt;Limitations&lt;br&gt;&amp; Unknowns&lt;/div&gt;&lt;/div&gt;
&lt;div class="extension-tile gridset-c animated bounceInDown"&gt;&lt;div&gt;COVID-19&lt;br&gt;Aftermath&lt;/div&gt;&lt;/div&gt;
&lt;/div&gt;



---
layout: false
class: full-screen hide-count

&lt;div class="grid-3-1"&gt;
&lt;div class="extension-tile gridset-a"&gt;&lt;div&gt;COVID-19&lt;br&gt;Genomic&lt;br&gt;Epidemiology&lt;/div&gt;&lt;/div&gt;
&lt;div class="extension-tile gridset-grey"&gt;&lt;div&gt;Biases&lt;br&gt;Limitations&lt;br&gt;&amp; Unknowns&lt;/div&gt;&lt;/div&gt;
&lt;div class="extension-tile gridset-grey"&gt;&lt;div&gt;COVID-19&lt;br&gt;Aftermath&lt;/div&gt;&lt;/div&gt;
&lt;/div&gt;




---

## Genomic Epidemiology        
  
.pad-left[  
  
- __Epidemiology__:    
  
    - study of the distribution and determinants of health-related outcomes in a specified population  

- __Molecular Epidemiology__:     
  
    -  joins understanding of disease at the molecular level with population-based study designs and approaches; links epidemiology with laboratory sciences  
    
- __Genomic Epidemiology__:     
  
    -  use of pathogen genomic data to determine the distribution and spread of an infectious disease in a specified population  
      
]  

.footnote-left[[CDC COVID-19 Genomic Epidemiology Toolkit 1.1](https://www.cdc.gov/amd/pdf/slidesets/toolkitmodule_1.1-508c.pdf); Lash TL et al _Modern Epidemiology (4th ed)_]  
  
  
  
---

## SARS-CoV-2 Genomic Epidemiology        
  
.pad-left[  
  
- __SARS-CoV-2 Genomes__:    
  
    - SARS-CoV-2 genome: ~ 30,000 nucleotides (~ 10,000 amino acids)  
    
    - infer relatedness from phylogenetic distance    
    
- __RNA Genome Sequencing__:     
  
    - extract RNA, reverse transcription, multiplex PCR, end repair &amp; ligation indexing  
    
    - short-read or long-read sequencing  
  
]  

.footnote-left[[CDC COVID-19 Genomic Epidemiology Toolkit 1.1](https://www.cdc.gov/amd/pdf/slidesets/toolkitmodule_1.1-508c.pdf)]  
  
  
  
---
  
## National &amp; Local Applications    
  
.pad-left[  
  
- **National**:  

    - monitor emergence and movement of new strains   
  
    - monitor trends after intervention (e.g., vaccination)   
  
- **Local**:   
    
    - investigate clusters for transmission (workplace, healthcare, etc)    
    
    - reveal unexpected clusters    
    
]  


.footnote-left[[CDC COVID-19 Genomic Epidemiology Toolkit 1.1](https://www.cdc.gov/amd/pdf/slidesets/toolkitmodule_1.1-508c.pdf); [Washington Post](https://www.washingtonpost.com/graphics/2020/health/coronavirus-genetic-code/)]  

  
  
---
exclude: false

## International Collaboration      
  
.center[
&lt;iframe src="https://gisaid.org/" width="95%" height="400px" data-external="1"&gt;&lt;/iframe&gt;
]

  
.footnote-left[https://gisaid.org/]  

  
---
exclude: false

## International Collaboration      
  
.center[
&lt;img src="figs/nextstrain_ncov_screenshot.png" width="60%" style="display: block; margin: auto;" /&gt;
]

  
.footnote-left[https://nextstrain.org/ncov/open/global/all-time]  

  

---

## Variants of Concern  

.center[
&lt;iframe src="https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-classifications.html" width="95%" height="400px" data-external="1"&gt;&lt;/iframe&gt;
]

  
.footnote-left[https://www.cdc.gov/coronavirus/2019-ncov/variants/variant-classifications.html]  

  

---
exclude: false

## Local Tracking        
  
.center[
&lt;iframe src="https://microb120.med.upenn.edu/data/SARS-CoV-2" width="95%" height="400px" data-external="1"&gt;&lt;/iframe&gt;
]

  
.footnote-left[https://microb120.med.upenn.edu/data/SARS-CoV-2]  
  


---
layout: false
class: full-screen hide-count

&lt;div class="grid-3-1"&gt;
&lt;div class="extension-tile gridset-grey"&gt;&lt;div&gt;COVID-19&lt;br&gt;Genomic&lt;br&gt;Epidemiology&lt;/div&gt;&lt;/div&gt;
&lt;div class="extension-tile gridset-b"&gt;&lt;div&gt;Biases&lt;br&gt;Limitations&lt;br&gt;&amp; Unknowns&lt;/div&gt;&lt;/div&gt;
&lt;div class="extension-tile gridset-grey"&gt;&lt;div&gt;COVID-19&lt;br&gt;Aftermath&lt;/div&gt;&lt;/div&gt;
&lt;/div&gt;


---

## Biases: SARS-CoV-2 "Viral Load"    
  
.pad-left[  
  
- The number of virions detectable in respiratory secretions varies between individuals and over the course of infection:    
  
    - high viral loads are required for most sequencing-preparation pipelines    
    
    - coverage/quality may be poor if viral loads are low    
  
- What are the consequences of excluding low-viral-load samples from genomic epidemiology surveillance?  
    
]    


---

.center[
&lt;img src="./figs//p_ct_distribution_ridgeplot_flip.png" width="80%" style="display: block; margin: auto;" /&gt;

]



---

.center[
&lt;img src="./figs/p_ct_distribution_ridgeplot_with_seqbar_flip.png" width="80%" style="display: block; margin: auto;" /&gt;

]


---

## Limitations: SARS-CoV-2 vs Population Immunity    
  
.pad-left[  
  
- Molecular evolution occurs in the context of increasingly complex and heterogeneous population immunity:    
  
    - waves of SARS-CoV-2 variants with different immune epitopes    
    
    - the introduction of vaccines    
  
- How best to measure SARS-CoV-2 molecular evolution in this context? (e.g., how best to perform "antigenic cartography"?)  
    
]    

.footnote-left[Mykytyn AZ et al _Science Immunology_ 2022]  


---

## SARS-CoV-2 Molecular Evolution    
  
.pad-left[  
  
- How do we measure SARS-CoV-2 genome change?  
  
    - daily, weekly, monthly?    
    
    - nucleotide, dN/dS, AA, gene?
    
    - how to pool within codons, genes?    

    - covariance across genome?

- How does genomic change relate to changes in incidence?  


    
]  


---

## Genomic Positional Diversity

.pad-left[

- Shannon diversity: 
    
  `$$H' = - \sum{ p_{i} * \log_{b}{(p_{i})} }$$`
    
&lt;center&gt;(note: typically natural log or base 2 are used)&lt;/center&gt;

- richness: how many nucleotide variants are there?
    
- evenness: how are variants distributed?


]


---

## Aside on Information Theory

.pad-left[

- Shannon diversity:

    `$$H' = - \sum{ p_{i} * \log_{b}{(p_{i})} }$$`
    
- Claude Shannon &amp; information entropy:

    `$$H(p) = - \sum{ p_{i} * \log_{b}{(p_{i})} }$$`
    
- **"The uncertainty contained in a probability distribution is the average log-probability of an event."**

]

.footnote[McElreath _Statistical Rethinking, 2nd_ 2020]


---
exclude: true
background-image: url(figs/scv2_snv_diversity_per_position_and_time_clear.gif)
background-size: contain


---
exclude: true
background-image: url(figs/scv2_snv_diversity_per_position_and_time_shadow.gif)
background-size: contain


---

&lt;img src="figs/scv2_snv_diversity_per_position_and_time_clear_color.gif" width="100%" style="display: block; margin: auto;" /&gt;


---

&lt;img src="figs/scv2_snv_diversity_per_position_and_time_shadow_color.gif" width="100%" style="display: block; margin: auto;" /&gt;


---

&lt;img src="figs/scv2_peptide_diversity_per_position_and_time_clear.gif" width="100%" style="display: block; margin: auto;" /&gt;


---

&lt;img src="figs/scv2_peptide_diversity_per_position_and_time_shadow.gif" width="100%" style="display: block; margin: auto;" /&gt;

---

&lt;img src="figs/p_phl_cases.png" width="100%" style="display: block; margin: auto;" /&gt;



---

&lt;img src="figs/p_lm_logdiff_peptide_color.png" width="100%" style="display: block; margin: auto;" /&gt;


---

&lt;img src="figs/p_lm_logdiff_peptide_color_tag.png" width="100%" style="display: block; margin: auto;" /&gt;



---
exclude: false

## The Canyon Hypothesis    
  
.pad-left[

- SARS-CoV-2 molecular evolution &amp; the "Canyon Hypothesis":

    - highly conserved cell entry mechanism mediated by the spike protein (S gene)

    - spike engages angiotensin converting enzyme 2 (ACE2) &amp; host proteases enable efficient membrane fusion between virions and target cells

    - a model by which sarbecoviruses are activated for fusion competency and interplay between humoral immunity and the molecular evolution

]


.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]




---
exclude: false

## The Canyon Hypothesis    
  
.pad-left[

- during circulation in populations with high rates of humoral immunity, viral entry proteins favor predominantly closed RBD configurations

- immediately after spillover into a population that lacks immunity, the newly emergent virus remains closely related to its ancestor

- during sustained transmission between seronegative individuals, wide transmission bottlenecks facilitate rapid emergence of variants that favor open RBD configurations to spread rapidly

- as humoral immunity expands, it gradually leads to a return to closed RBDs as repeat exposures facilitate the affinity maturation of expansive antibody repertoires that are disproportionately costly to open RBD configurations

]


.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]




---

## The Canyon Hypothesis    

.center[
&lt;img src="./figs/canyon_hypothesis_rbd_open_closed.png" width="80%" style="display: block; margin: auto;" /&gt;

]

.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]


---
exclude: false

## The Canyon Hypothesis    
  
.pad-left[

- most changes in S prior to the emergence of the Omicron variant appear to have been driven by selection for improved transmission between immunologically naive hosts

- S:D614G spike mutation improved ACE2 affinity but also made RBDs were more likely to hold the open conformation

- cryo-EM studies of the Omicron S protein suggest that while the Delta Spike predominantly occupies conformations with 1 or more RBDs open simultaneously, the Omicron Spike appears to prefer conformations with 0 or 1 open RBD

- antigenicity of stabilizing elements &amp; tendency of primary immune responses to generate a limited repertoire of antibodies may explain the selection for open RBDs early in the pandemic &amp; shift in the selective landscape that led to the Omicron variant’s emergence and rapid sweep

]


.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]



---
exclude: true

.center[
&lt;img src="./figs//p_ct_distribution_ridgeplot_flip.png" width="80%" style="display: block; margin: auto;" /&gt;

]



---
exclude: true

.center[
&lt;img src="./figs/p_ct_distribution_ridgeplot_with_seqbar_flip.png" width="80%" style="display: block; margin: auto;" /&gt;

]



---

.center[
&lt;img src="./figs/alpha_delta_omicron_ct_comparison.png" width="80%" style="display: block; margin: auto;" /&gt;

]



---

## Unknowns: Where did 'weird' Omicron come from?

.center[
&lt;img src="./figs/new_long_branch_screenshot.png" width="80%" style="display: block; margin: auto;" /&gt;

]

.footnote-left[[Kupferschmidt K *Science* 1 Dec 2021](https://www.science.org/content/article/where-did-weird-omicron-come)]  

---

## Unknowns: Where did 'weird' Omicron come from?

.center[
&lt;img src="./figs/three_explanations_screenshot.png" width="80%" style="display: block; margin: auto;" /&gt;

]

.footnote-left[[Kupferschmidt K *Science* 1 Dec 2021](https://www.science.org/content/article/where-did-weird-omicron-come)]  




---

## Pennsylvania Deer

.center[
&lt;img src="./figs/mbio.02101-22-f005.gif" width="80%" style="display: block; margin: auto;" /&gt;

]

.footnote-left[[Marques AD et al *mBio* 2022](http://dx.doi.org/10.1128/mbio.02101-22)]  




---
layout: false
class: full-screen hide-count

&lt;div class="grid-3-1"&gt;
&lt;div class="extension-tile gridset-grey"&gt;&lt;div&gt;COVID-19&lt;br&gt;Genomic&lt;br&gt;Epidemiology&lt;/div&gt;&lt;/div&gt;
&lt;div class="extension-tile gridset-grey"&gt;&lt;div&gt;Biases&lt;br&gt;Limitations&lt;br&gt;&amp; Unknowns&lt;/div&gt;&lt;/div&gt;
&lt;div class="extension-tile gridset-c"&gt;&lt;div&gt;COVID-19&lt;br&gt;Aftermath&lt;/div&gt;&lt;/div&gt;
&lt;/div&gt;




---

## Lessons &amp; Directions After COVID-19      
  
.pad-left[  
  
- What has come of this unprecedented investment in molecular/genomic epidemiology?  
  
    - unprecedented pathogen sequencing capacity    
    
    - novel methods of population surveillance: wastewater surveillance    

    - political animus against CDC &amp; resulting budget cuts  
    
]  

---

## Ongoing SARS-CoV-2 Variant Tracking   

.center[
&lt;iframe src="https://covid.cdc.gov/covid-data-tracker/#variant-proportions" width="95%" height="400px" data-external="1"&gt;&lt;/iframe&gt;
]

.footnote[[https://covid.cdc.gov/covid-data-tracker/#variant-proportions](https://covid.cdc.gov/covid-data-tracker/#variant-proportions)]


---

## Wastewater SARS-CoV-2 Surveillance     

.center[
&lt;iframe src="https://covid.cdc.gov/covid-data-tracker/#wastewater-surveillance" width="95%" height="400px" data-external="1"&gt;&lt;/iframe&gt;
]

.footnote[[https://covid.cdc.gov/covid-data-tracker/#wastewater-surveillance](https://covid.cdc.gov/covid-data-tracker/#wastewater-surveillance)]



---

## Centers for Pathogen Genomics       

.center[
&lt;iframe src="https://www.cdc.gov/media/releases/2022/p0920-PGCoE-network.html" width="95%" height="400px" data-external="1"&gt;&lt;/iframe&gt;
]

.footnote[[https://www.cdc.gov/media/releases/2022/p0920-PGCoE-network.html](https://www.cdc.gov/media/releases/2022/p0920-PGCoE-network.html)]



---

## NWSS: National Wastewater Surveillance System         

.center[
&lt;iframe src="https://www.cdc.gov/nwss/index.html" width="95%" height="400px" data-external="1"&gt;&lt;/iframe&gt;
]

.footnote[[https://www.cdc.gov/nwss/index.html](https://www.cdc.gov/nwss/index.html)]



---

## CDC Budget Cuts        

.pad-left[  
  
- CDC and NIH have become political targets:   
  
    - $1.3 billion (~18%) cut from CDC as part of debt ceiling agreement      
    
    - some budget proposals for FY2024 include a total $11.5 billion ($2.8 billion increase from FY2023)     

    - other budget proposals for FY2024 include additional 20-30% cuts  
    
]  



---

## Lessons &amp; Directions After COVID-19      
  
.pad-left[  
  
- What has come of this unprecedented investment in molecular/genomic epidemiology?    
  
    - unprecedented pathogen sequencing capacity    
    
    - novel methods of population surveillance: wastewater surveillance    

    - political animus against CDC &amp; resulting budget cuts  

- **Will re-investment come before or after the next pandemic?**  

    
]  


---
exclude: true

## Acknowledgements    

.pull-left[
- __ARES Group @ Penn__
Laurel Glaser, Kyle Rodino, Sean Loughrey, Laura Cowden,   
Magda Wernovsky, Jessie Gunoskey, Jennifer Han, Erik Clarke, Michael David,
Matt Ziegler, Lauren Dutcher, Ebbing Lautenbach, Jim Harrigan  

- __Bushman Laboratory @ Penn__
Arwa Abbas, Aoife Roche, Andrew Marques,
Aubrey Bailey, John Everett,
Rick Bushman

- __PennCHOP Microbiome Program__
Lisa Mattei, Casey Hofstaedter, Huanjia Zhang, Kyle Bittinger

]


.pull-right[
- __Collman Laboratory @ Penn__
Ize Imai, Aurea Simon Soro, John McGinniss, Ron Collman

- __Division of ID @ Penn__
Ian Frank, Pablo Tebas, Robert Gross, Emily Blumberg

- __Rutgers University &amp; Penn DBEI__
Jason Roy, Arman Oganisian
  
- __CDC Prevention Epicenters__
Clifford McDonald, Alison Laufer Halpin
  
- __Funding__
&lt;u&gt;CDC&lt;/u&gt;: BAAs 200-2016-91964, 200-2018-02919, 200-2021-10986,
200-2021-10986 &amp; &lt;u&gt;NIAID&lt;/u&gt;: K23 AI121485

]

.center[
### brendank@pennmedicine.upenn.edu
]


---
class: full-screen hide-count

&lt;div class="grid-3-1"&gt;
&lt;div class="extension-tile gridset-a animated flipInY"&gt;&lt;div&gt;COVID-19&lt;br&gt;Genomic&lt;br&gt;Epidemiology&lt;/div&gt;&lt;/div&gt;
&lt;div class="extension-tile gridset-b animated flipInX"&gt;&lt;div&gt;Biases&lt;br&gt;Limitations&lt;br&gt;&amp; Unknowns&lt;/div&gt;&lt;/div&gt;
&lt;div class="extension-tile gridset-c animated bounceInDown"&gt;&lt;div&gt;COVID-19&lt;br&gt;Aftermath&lt;/div&gt;&lt;/div&gt;
&lt;/div&gt;




---
class: middle center hide-count
background-image: url(img/sarscov2_cdc_blur_crop.png)
background-size: cover

.title-subtext[Questions?]



---
exclude: true
class: middle center hide-count
background-image: url(img/sarscov2_cdc_blur_crop.png)
background-size: cover

.title-subtext[Questions?]

.callout-url-bottom[  
.fade-in[&lt;span style="font-size:0.5em; color:white"&gt;slides ↓ &lt;/span&gt;]  
&lt;span style="link-color:white"&gt;[bjklab.org](http://www.bjklab.org)&lt;/span&gt;  
]  


    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="libs/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"ratio": "16:9",
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false,
"slideNumberFormat": "%current%"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
