---
title: "SARS-CoV-2: Molecular Evolution & Prediction"
subtitle: "<hr>"
author: "Brendan J. Kelly, MD, MS"
institute: "Infectious Diseases & Epidemiology<br><br>![](penn_logos/Penn-Shield-Logotype-White-over-Blue-RGB-Expanded.svg){width=160px}" 
date: "`r format(Sys.Date(), '%d %B %Y')`"
output:
  xaringan::moon_reader:
    css: xaringan-themer-inferno-grids-fade.css
    lib_dir: libs
    chakra: libs/remark-latest.min.js
    seal: false
    nature:
      ratio: 16:9
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
      slideNumberFormat: "%current%"

---
exclude: true

```{r setup, include=FALSE, echo=FALSE, message=FALSE, error=FALSE, include=FALSE, comment=FALSE}
options(htmltools.dir.version = FALSE)
xaringan::summon_remark()
library(tidyverse)
library(gt)

```

```{r xaringan-tile-view, echo=FALSE, message=FALSE, error=FALSE, include=FALSE, comment=FALSE}
xaringanExtra::use_tile_view()
xaringanExtra::use_webcam()
xaringanExtra::use_tachyons()
xaringanExtra::use_animate_css()
xaringanExtra::use_share_again()
#xaringanExtra::use_scribble(pen_color = "#56106E")
```

---
class: top left hide-count
background-image: url(img/sarscov2_cdc_blur_crop.png)
background-size: cover

.move-top20[
.title-text[<span style='font-size:1.0em'>SARS-CoV-2</span>]  
.title-subtext[<span style='font-size:1.0em'>Molecular Evolution & Prediction</span>]  
]  
  
.callout-url-leftcorner[
.title-nametext[
Brendan J. Kelly, MD, MS  
Infectious Diseases, Epidemiology & Microbiology  
University of Pennsylvania  
`r format(Sys.Date(), '%d %B %Y')`   
]
]

.footnote-right[<span style='color: white'>photo: CDC @unsplash</span>]  



---

## Disclosures  

.pad-left[  

- No conflicts of interest.  

- Research supported by:

    - NIAID K23 AI121485  
    
    - CDC BAA 200-2021-10986   

]



---
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-a animated flipInY"><div>SARS-CoV-2<br>Molecular<br>Evolution</div></div>
</div>


---
layout: false
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-a"><div>SARS-CoV-2<br>Molecular<br>Evolution</div></div>
<div class="extension-tile gridset-b animated flipInX"><div>The Canyon<br>Hypothesis</div></div>
</div>


---
layout: false
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-a"><div>SARS-CoV-2<br>Molecular<br>Evolution</div></div>
<div class="extension-tile gridset-b"><div>The Canyon<br>Hypothesis</div></div>
<div class="extension-tile gridset-c animated bounceInDown"><div>Cycle Threshold<br>& Prediction</div></div>
</div>



---
layout: false
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-a"><div>SARS-CoV-2<br>Molecular<br>Evolution</div></div>
<div class="extension-tile gridset-grey"><div>The Canyon<br>Hypothesis</div></div>
<div class="extension-tile gridset-grey"><div>Cycle Threshold<br>& Prediction</div></div>
</div>




---

## SARS-CoV-2 Molecular Evolution    
  
.pad-left[  
  
- How do we measure SARS-CoV-2 genome change?  
  
    - daily, weekly, monthly?    
    
    - nucleotide, dN/dS, AA, gene?
    
    - how to pool within codons, genes?    

    - covariance across genome?

- How does genomic change relate to changes in incidence?  


    
]  


---

## Genomic Positional Diversity

.pad-left[

- Shannon diversity: 
    
  $$H' = - \sum{ p_{i} * \log_{b}{(p_{i})} }$$
    
<center>(note: typically natural log or base 2 are used)</center>

- richness: how many nucleotide variants are there?
    
- evenness: how are variants distributed?


]


---

## Aside on Information Theory

.pad-left[

- Shannon diversity:

    $$H' = - \sum{ p_{i} * \log_{b}{(p_{i})} }$$
    
- Claude Shannon & information entropy:

    $$H(p) = - \sum{ p_{i} * \log_{b}{(p_{i})} }$$
    
- **"The uncertainty contained in a probability distribution is the average log-probability of an event."**

]

.footnote[McElreath _Statistical Rethinking, 2nd_ 2020]


---
exclude: true
background-image: url(figs/scv2_snv_diversity_per_position_and_time_clear.gif)
background-size: contain


---
exclude: true
background-image: url(figs/scv2_snv_diversity_per_position_and_time_shadow.gif)
background-size: contain


---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/scv2_snv_diversity_per_position_and_time_clear_color.gif")

```


---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/scv2_snv_diversity_per_position_and_time_shadow_color.gif")

```


---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/scv2_peptide_diversity_per_position_and_time_clear.gif")

```


---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/scv2_peptide_diversity_per_position_and_time_shadow.gif")

```

---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/p_phl_cases.png")

```



---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/p_lm_logdiff_peptide_color.png")

```


---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "figs/p_lm_logdiff_peptide_color_tag.png")

```



---
layout: false
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-grey"><div>SARS-CoV-2<br>Molecular<br>Evolution</div></div>
<div class="extension-tile gridset-b"><div>The Canyon<br>Hypothesis</div></div>
<div class="extension-tile gridset-grey"><div>Cycle Threshold<br>& Prediction</div></div>
</div>



---
exclude: false

## The Canyon Hypothesis    
  
.pad-left[

- SARS-CoV-2 molecular evolution & the "Canyon Hypothesis":

    - highly conserved cell entry mechanism mediated by the spike protein (S gene)

    - spike engages angiotensin converting enzyme 2 (ACE2) & host proteases enable efficient membrane fusion between virions and target cells

    - a model by which sarbecoviruses are activated for fusion competency and interplay between humoral immunity and the molecular evolution

]


.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]




---
exclude: false

## The Canyon Hypothesis    
  
.pad-left[

- during circulation in populations with high rates of humoral immunity, viral entry proteins favor predominantly closed RBD configurations

- immediately after spillover into a population that lacks immunity, the newly emergent virus remains closely related to its ancestor

- during sustained transmission between seronegative individuals, wide transmission bottlenecks facilitate rapid emergence of variants that favor open RBD configurations to spread rapidly

- as humoral immunity expands, it gradually leads to a return to closed RBDs as repeat exposures facilitate the affinity maturation of expansive antibody repertoires that are disproportionately costly to open RBD configurations

]


.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]




---

## The Canyon Hypothesis    

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs/canyon_hypothesis_rbd_open_closed.png")

```

]

.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]


---
exclude: false

## The Canyon Hypothesis    
  
.pad-left[

- most changes in S prior to the emergence of the Omicron variant appear to have been driven by selection for improved transmission between immunologically naive hosts

- S:D614G spike mutation improved ACE2 affinity but also made RBDs were more likely to hold the open conformation

- cryo-EM studies of the Omicron S protein suggest that while the Delta Spike predominantly occupies conformations with 1 or more RBDs open simultaneously, the Omicron Spike appears to prefer conformations with 0 or 1 open RBD

- antigenicity of stabilizing elements & tendency of primary immune responses to generate a limited repertoire of antibodies may explain the selection for open RBDs early in the pandemic & shift in the selective landscape that led to the Omicron variant’s emergence and rapid sweep

]


.footnote[Wolf et al *mBio* 2022; Rossman et al *J Biol Chem* 1989]



---
layout: false
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-grey"><div>SARS-CoV-2<br>Molecular<br>Evolution</div></div>
<div class="extension-tile gridset-grey"><div>The Canyon<br>Hypothesis</div></div>
<div class="extension-tile gridset-c"><div>Cycle Threshold<br>& Prediction</div></div>
</div>




---

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs//p_ct_distribution_ridgeplot_flip.png")

```

]



---

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs/p_ct_distribution_ridgeplot_with_seqbar_flip.png")

```

]



---

.center[
```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "./figs/alpha_delta_omicron_ct_comparison.png")

```

]



---
exclude: true

## Acknowledgements    

.pull-left[
- __ARES Group @ Penn__
Laurel Glaser, Kyle Rodino, Sean Loughrey, Laura Cowden,   
Magda Wernovsky, Jennifer Han, Erik Clarke, Michael David,
Matt Ziegler, Lauren Dutcher, Ebbing Lautenbach, Jim Harrigan, Hatem Abdallah

- __Bushman Laboratory @ Penn__
Arwa Abbas, Aoife Roche, Andrew Marques,
Aubrey Bailey, Jacob Leiby,
Rick Bushman

- __PennCHOP Microbiome Program__
Lisa Mattei, Casey Hofstaedter, Huanjia Zhang, Kyle Bittinger

]


.pull-right[
- __Collman Laboratory @ Penn__
Ize Imai, Aurea Simon Soro, John McGinniss, Ron Collman

- __Division of ID @ Penn__
Ian Frank, Pablo Tebas, Robert Gross, Emily Blumberg

- __Rutgers University & Penn DBEI__
Jason Roy, Arman Oganisian
  
- __CDC Prevention Epicenters__
Clifford McDonald, Alison Laufer Halpin
  
- __Funding__
<u>CDC</u>: BAAs 200-2016-91964, 200-2018-02919, 200-2021-10986,
200-2021-10986 & <u>NIAID</u>: K23 AI121485

]

.center[
### brendank@pennmedicine.upenn.edu
]



---
class: full-screen hide-count

<div class="grid-3-1">
<div class="extension-tile gridset-a animated flipInY"><div>SARS-CoV-2<br>Molecular<br>Evolution</div></div>
<div class="extension-tile gridset-b animated flipInX"><div>The Canyon<br>Hypothesis</div></div>
<div class="extension-tile gridset-c animated bounceInDown"><div>Cycle Threshold<br>& Prediction</div></div>
</div>




---
class: middle center hide-count
background-image: url(img/sarscov2_cdc_blur_crop.png)
background-size: cover

.title-subtext[Questions?]



---
exclude: true
class: middle center hide-count
background-image: url(img/sarscov2_cdc_blur_crop.png)
background-size: cover

.title-subtext[Questions?]

.callout-url-bottom[  
.fade-in[<span style="font-size:0.5em; color:white">slides ↓ </span>]  
<span style="link-color:white">[bjklab.org](http://www.bjklab.org)</span>  
]  


